test/validation-error-handling.test.js

```diff
@@ -831,3 +831,72 @@ test('plugin override', async (t) => {
   })
   t.assert.strictEqual(response5.statusCode, 400)
 })
+
+test('async preValidation with custom validator should trigger error handler when validator throws', async (t) => {
+  t.plan(8)
+
+  const fastify = Fastify()
+
+  // Set up error handler
+  fastify.setErrorHandler((error, request, reply) => {
+    t.assert.ok(error instanceof Error, 'error should be an Error instance')
+    t.assert.strictEqual(error.message, 'Custom validation failed')
+    reply.status(500).send({ error: error.message })
+  })
+
+  // Route with custom validator that throws
+  const routeDefinition = {
+    schema: {
+      body: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' }
+        }
+      }
+    },
+    validatorCompiler: ({ schema, method, url, httpPart }) => {
+      return function (data) {
+        // This custom validator throws an error instead of returning {error}
+        throw new Error('Custom validation failed')
+      }
+    },
+    handler (request, reply) {
+      t.assert.fail('Handler should not be called')
+      reply.send({ success: true })
+    }
+  }
+
+  fastify.register(async function plugin (app, opts) {
+    // Add sync preValidation hook
+    app.addHook('preValidation', (request, reply, next) => { next() })
+    app.post('/sync', routeDefinition)
+  })
+
+  fastify.register(async function plugin (app, opts) {
+    // Add async preValidation hook
+    fastify.addHook('preValidation', async (request, reply) => {
+      await new Promise(resolve => setImmediate(resolve))
```

Comment:
Why use `setImmediate` here? Is it only for syntactic purposes?
If so, I’d prefer removing it, it might makes the reader overthink its intent.

```suggestion
await Promise.resolve('ok')
```